runtime/io.h

```diff
@@ -133,43 +133,71 @@ void IONAME(SetSign)(Cookie, const char *, std::size_t);
 // Any item can be transferred by means of a descriptor; unformatted
 // transfers to/from contiguous blocks can avoid the descriptor; and there
 // are specializations for the common scalar types.
-// Functions with Boolean results return false when the I/O statement
-// has encountered an error or end-of-file/record condition; the others
-// will return a zero value.
+// These functions return false when the I/O statement has encountered an error
+// or end-of-file/record condition.  Once the statement has encountered
+// an error, following items will be ignored; but compiled code should
+// check for errors and avoid the following items when a READ statement
+// might have data dependences between items, as in
```

Comment:
While checking to verify if this io-implied-do READ is allowed (it looks like it is - 12.6.3p2), I noticed the following case in nearby Note 3.  Can descriptors handle vector subscripts that don't have repeat elements?

INTEGER A(100), J(100)
READ*, A(J)

Reply:
No, descriptors cannot represent vector valued subscripts.  Such objects in data lists will have to be expanded into implied DO loops in the compiled code.